src/ucp/api/ucp.h

```diff
@@ -2481,15 +2482,28 @@ ucp_atomic_fetch_nb(ucp_ep_h ep, ucp_atomic_fetch_op_t opcode,
  *
  * @param [in]  request     Non-blocking request to check.
  *
- * @param [out] info        If request is in completed state, it is
- *                          filled with the details about the message.
+ * @return Error code as defined by @ref ucs_status_t
+ */
+ucs_status_t ucp_request_check_status(void *request);
+
+
+/**
+ * @ingroup UCP_COMM
+ * @brief Check the status and currently available state of non-blocking request
+ *        returned from @ref ucp_tag_recv_nb routine.
  *
- * @note The @p info parameter is relevant for receive operations only. It is
- * left uninitialized in case of send operation.
+ * This routine checks the state and returns current status of the request
+ *      returned from @ref ucp_tag_recv_nb routine or the user allocated request
+ *      for @ref ucp_tag_recv_nbr. Any value different from UCS_INPROGRESS means
+ *      that the request is in a completed state.
+ *
+ * @param [in]  request     Non-blocking request to check.
+ * @param [out] info        It is filled with the details about the message
+ *                          available at the moment of calling.
  *
  * @return Error code as defined by @ref ucs_status_t
  */
-ucs_status_t ucp_request_test(void *request, ucp_tag_recv_info_t *info);
+ucs_status_t ucp_tag_recv_request_test(void *request, ucp_tag_recv_info_t *info);
```

Comment:
The detailed description for ucp_tag_recv_nbr still references ucp_request_test, so needs to be updated accordingly.